enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,252 @@
+* An openshift-krew-index with supported artifacts could hold information about CLIs supported by OpenShift.
+* With `krew index add https://someother-third-party-index` we won't limit cluster-admins from adding their own index with whatever plugins they want. Only the openshift-krew-index will be supported.
+
+Users will continue to have the option to download binaries from either the central location where artifacts are published or from the console.
+This proposal adds the mechanism to install from images built from the same artifacts using `oc`. All options will offer the same binaries,
+ensured through sha256 signatures.
+
+For the initial implementation, supported CLIs will create ClusterCLI custom resources.  The CLI will be installed to a user's home directory with
+the same logic as `oc image extract` and from each CLI's artifacts image (see oc, cli-artifacts image for reference).
+Through images and ImageContentSourcePolicy, CLIs will be available to disconnected environments. Currently, the console provides
+oc binaries by running a webserver pod in the console namespace with a route.  The cli-manager will take over running this webserver, this will
+be included in the CRD controller that will create the core ClusterCLI custom resources (oc, odo, and helm, the 3 CLIs currently available
+through the consoleclidownloads mechansism).
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs available for OpenShift and related services so that I can discover, install and list them
+The user will invoke the following commands:
+
+* `oc tools list` will access the connected OpenShift cluster, if connected,  and will retrieve information about available CLIs (ClusterCLI CRs)
+* `oc tools install odo` will install `odo` to a user's home directory, as a standalone binary.
+
+##### Example
+```text
+$ oc tools list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+```
+
+#### Story 2
+
+As a user, I want to access various CLIs from the OpenShift Console Web Terminal so that I can use the
+services available in the cluster. The CLIs versions should match the those of the operators deployed
+on the cluster.
+
+Currently, users can access `consoleclidownloads` custom resources for helm, oc, and odo either from oc with `oc get consoleclidownloads` or through
+the `downloads -n openshift-console` route. This access will still be available, through the updated ClusterCLI custom resources. The ClusterCLI custom
```

Comment:
Does this mean that newer versions of oc will use the new CRD instead of the old mechanism? Is there a version skew concern with that change, or will the old mechanism work for old clients?

Reply:
The old mechanism isn't going away yet (maybe never). The current `oc` command to extract binaries from those CRs isn't changing either. This would add a new subcommand to `oc` that will be used exclusively for the new CRD.